File: kutuphane-rezervasyon-frontend/gatling-tests/src/test/scala/simulations/BasicSmokeTest.scala
package simulations

import io.gatling.core.Predef._
import io.gatling.http.Predef._
import scala.concurrent.duration._

class BasicSmokeTest extends Simulation {

  val baseUrl = "http://localhost:8080"
  
  val httpProtocol = http
    .baseUrl(baseUrl)
    .acceptHeader("application/json")
    .contentTypeHeader("application/json")

  // Basit Smoke Test - Tüm endpoint'lerin çalıştığını kontrol eder
  val smokeTest = scenario("Basic Smoke Test")
    .exec(http("Health Check - Rooms")
      .get("/api/rooms")
      .check(status.in(200, 401))) // 401 da kabul edilebilir (auth gerekli)
    .pause(1)
    .exec(http("Health Check - Equipment")
      .get("/api/equipment")
      .check(status.in(200, 401)))
    .pause(1)
    .exec(http("Health Check - Time Slots")
      .get("/api/timeslots")
      .check(status.in(200, 401)))
    .pause(1)
    .exec(http("Login Test")
      .post("/api/auth/login")
      .body(StringBody("""{"email":"admin@example.com","password":"admin123"}""")).asJson
      .check(status.is(200))
      .check(jsonPath("$.token").exists))

  setUp(
    smokeTest.inject(atOnceUsers(1))
  ).protocols(httpProtocol)
}

File: kutuphane-rezervasyon-frontend/gatling-tests/src/test/java/simulations/BasicSmokeTest.java
package simulations;

import io.gatling.javaapi.core.*;
import io.gatling.javaapi.http.*;

import static io.gatling.javaapi.core.CoreDsl.*;
import static io.gatling.javaapi.http.HttpDsl.*;

public class BasicSmokeTest extends Simulation {

    HttpProtocolBuilder httpProtocol = http
        .baseUrl("http://localhost:8080")
        .acceptHeader("application/json")
        .contentTypeHeader("application/json")
        .header("Origin", "http://localhost:4201");

    ScenarioBuilder scn = scenario("Basic Smoke Test")
        .exec(http("Health Check - Rooms")
            .get("/api/rooms")
            .check(status().is(200)))
        .pause(1)
        .exec(http("Health Check - Equipment")
            .get("/api/equipment")
            .check(status().is(200)))
        .pause(1)
        .exec(http("Health Check - Time Slots")
            .get("/api/timeslots")
            .check(status().is(200)))
        .pause(1)
        .exec(http("Login Test")
            .post("/api/auth/login")
            .body(StringBody("{\"email\":\"root@kutuphane.com\",\"password\":\"root123\"}"))
            .asJson()
            .check(status().is(200))
            .check(jsonPath("$.userId").exists())
            .check(jsonPath("$.role").is("ADMIN")));

    {
        setUp(
            scn.injectOpen(atOnceUsers(1))
        ).protocols(httpProtocol);
    }
}
